flaml/nlp/autotransformers.py

```diff
@@ -0,0 +1,901 @@
+        try:
+            ckpt_json = json.load(open(ckpt_dir))
+            return ckpt_json["best_ckpt"]
+        except FileNotFoundError as err:
+            logger.error("Saved checkpoint not found. Please make sure checkpoint is stored under {}".format(ckpt_dir))
+            raise err
+
+    def _set_metric(self, metric_name, metric_mode_name):
+        default_metric, default_mode, all_metrics, all_modes = get_default_and_alternative_metric(self._dataset_name[0], self._subdataset_name, metric_name, metric_mode_name)
+        _variable_override_default_alternative(logger, self, "metric_name", default_metric, all_metrics, metric_name)
+        _variable_override_default_alternative(logger, self, "metric_mode_name", default_mode, all_modes, metric_mode_name)
+
+    def fit(self,
+            train_dataset,
+            eval_dataset,
+            resources_per_trial,
+            wandb_key,
+            metric_name = None,
+            metric_mode_name = "max",
+            search_algo_name= None,
+            ckpt_per_epoch=1,
+            fp16 = True,
+            search_space_path = None,
+            scheduler_name=None,
+            verbose = 1,
+            search_algo_args_mode = "default",
+            hpo_searchspace_mode = None,
+            num_sample_time_budget_mode = "custom",
+            custom_num_samples = None,
+            custom_time_budget = None,
+            time_as_grid = None,
+            **custom_search_algo_args):
+        '''Fine tuning the huggingface using the hpo setting
+
+        Args:
+            train_dataset:
+                the training data of type datasets.Dataset, loaded from datasets.load_dataset
+            eval_dataset:
+                the validation data of type datasets.Dataset, loaded from datasets.load_dataset
+            metric_name:
+                A string of the dataset name or a function,
+                e.g., 'accuracy', 'f1', 'loss',
+                if passing a customized dataset function, the function needs to
+                have the follwing signature:
+
+                .. code-block:: python
+
+                    def custom_metric(X_test, y_test, estimator, labels,
+                     X_train, y_train, weight_test=None, weight_train=None):
+                        return metric_to_minimize, metrics_to_log
+
+                which returns a float number as the minimization objective,
+                and a tuple of floats as the metrics to log
+            metric_mode_name:
+                A string of the mode name,
+                e.g., "max", "min", "last", "all"
+            resources_per_trial:
+                A dict showing the resources used by each trial,
+                e.g., {"gpu": 4, "cpu": 4}
+            wandb_key:
+                The hash code for wandb
+            search_algo_name:
+                The search algoritihm for AutoHF()
+                e.g., "blendsearch" "cfo" "bo"
```

Comment:
'blendsearch' or 'BlendSearch'? Be consistent with line 565. 
'cfo' is not handled. 
'bo' is not specific enough.